router/core/request_context_fields.go

```diff
@@ -68,29 +87,89 @@ func NewDurationLogField(val time.Duration, attribute config.CustomAttribute) za
 	return zap.Skip()
 }
 
-func AccessLogsFieldHandler(attributes []config.CustomAttribute, err any, request *http.Request, responseHeader *http.Header) []zapcore.Field {
+func RouterAccessLogsFieldHandler(
```

Comment:
Should we really create functions for this? I asked above if we really need to this in isolation or if we can test the full behaviour through an integration test. If yes, we could inline those functions.

Reply:
We use `run expression with an error` to check the error type (when I tried it in the integration test, strings were returned so we couldnt really check the type). but on top of that there are a few existing tests that relies on this function being isolated.